core-bundle/src/Resources/contao/library/Contao/Model.php

```diff
@@ -1240,7 +1240,10 @@ protected static function buildCountQuery(array $arrOptions)
 	 */
 	protected static function createModelFromDbResult(Database\Result $objResult)
 	{
-		return new static($objResult);
+		/** @var static $strClass */
```

Comment:
```suggestion
		/** @var class-string<static> $strClass */
```
PHPStan would support this AFAIK, not sure about PHPStorm.

Reply:
PHPStorm does not (yet). However, it would support the following:

```
/**
 * @var static $strClass
 * @phpstan-var class-string<static> $strClass
 */
```

We should create a separate PR for this, though, because Contao 4.4 does not even use PHPStan.

Reply:
See #2080